src/go/rpk/pkg/cli/cmd/cloud/login.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package cloud
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/cmd/cloud/auth"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/cmd/cloud/cloudcfg"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newLoginCommand(fs afero.Fs) *cobra.Command {
+	var params cloudcfg.Params
+	cmd := &cobra.Command{
+		Use:   "login",
+		Short: "Login to the Redpanda cloud",
+		Args:  cobra.ExactArgs(0),
+		Long: `Login to the Redpanda cloud
+
+This command checks for an existing token and, if present, ensures it is still
+valid. If no token is found or the token is no longer valid, this command will
+login and save your token.
+
+You may use any of the following methods to pass the cloud credentials to rpk:
+
+CONFIG FILE
+
+Store them in '$HOME/.config/rpk/__cloud.yaml' :
+  client_id:
+  client_secret
+
+FLAGS
+
+Use --client-id and --client-secret flags:
+
+  rpk cloud login --client-id <id> --client-secret <secret>
+
+ENVIRONMENT VARIABLES
+
+Use CLOUD_CLIENT_ID and CLOUD_CLIENT_SECRET environment flags
+
+  CLOUD_CLIENT_ID=<id> CLOUD_CLIENT_SECRET=<secret> rpk cloud login
+
+If neither of the above methods are used, rpk prompts for the credentials and
+store them in $HOME/.config/rpk/__cloud.yaml once the login succeeds.
+
+All commands in the cloud plugin check for / load / refresh a token as
+necessary, so this command is not really necessary to run directly.
+`,
+		Run: func(cmd *cobra.Command, _ []string) {
+			cfg, err := params.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			_, err = auth.LoadFlow(cmd.Context(), fs, cfg)
+			if e := (*auth.BadClientTokenError)(nil); errors.As(err, &e) {
+				out.Die("unable to login into Redpanda Cloud: %v; You can clear your client credentials with 'rpk cloud logout --clear-credentials'", err)
```

Comment:
I think we should format this just a little bit more -- we're matching on this specific type so that we can better format some suggestion. wdyt of:

```
`
Unable to login into Redpanda Cloud: %v.

You may need to clear your client ID and secret with 'rpk cloud logout --clear-credentials',
and then re-specify the client credentials next time you log in.
`
```
Basically, split this message up into multiple lines.